src/src/SceneNode.cpp

```diff
@@ -196,4 +198,31 @@ namespace mineola {
         return node.Name() == name_str;
       });
   }
+
+AABB SceneNode::ComputeAABB() {
+  AABB bbox{glm::vec3(std::numeric_limits<float>::max()),
+    glm::vec3(std::numeric_limits<float>::lowest())};
+
+  DFTraverse([&bbox](SceneNode &node) {
+    bool valid_aabb = false;
+    AABB bbox_node{glm::vec3(std::numeric_limits<float>::max()),
+      glm::vec3(std::numeric_limits<float>::lowest())};
+    for (auto renderable : node.Renderables()) {
+      // check min and max vertex is valid or not
+      if (renderable->Bbox()) {
+        bbox_node.Combine(*renderable->Bbox());
+        valid_aabb = true;
+      }
+    }
+
+    if (valid_aabb) {
+      auto model_mat = glm::scale(node.Rbt().ToMatrix(), node.Scale());
+      bbox_node.Transform(model_mat);
+      bbox.Combine(bbox_node);
+    }
+  });
+
+  return bbox;
+}
```

Comment:
std::optional<AABB> result;
for (auto &child: children) {
  auto aabb = child.ComputeAABB();
  if (aabb) {
    if (result) {
      result.Combine(aabb);
    } else {
      result = aabb;
    }
  }
}

if (result) {
    // apply model mat
}

return result;